more_itertools/more.py

```diff
@@ -166,17 +185,17 @@ def collate(*iterables, **kwargs):
     If the elements of the passed-in iterables are out of order, you might get
     unexpected results.
 
+    If neither of the keyword arguments are specified, or if the Python version
+    is 3.5 or greater, this function delegates to ``heapq.merge()``.
+
     """
-    key = kwargs.pop('key', lambda a: a)
-    reverse = kwargs.pop('reverse', False)
+    if not kwargs:
+        return merge(*iterables)
 
-    min_or_max = partial(max if reverse else min, key=lambda a_b: a_b[0])
-    peekables = [peekable(it) for it in iterables]
-    peekables = [p for p in peekables if p]  # Kill empties.
-    while peekables:
-        _, p = min_or_max((key(p.peek()), p) for p in peekables)
-        yield next(p)
-        peekables = [x for x in peekables if x]
+    if (version_info[0] >= 3) and (version_info[1] >= 5):
```

Comment:
For further efficiency, we could avoid the version check on every invocation by doing it once, at import time.